crates/ruff_db/src/diagnostic.rs

```diff
@@ -73,6 +74,47 @@ impl std::fmt::Display for DisplayDiagnostic<'_> {
     }
 }
 
+#[salsa::accumulator]
+pub struct CompileDiagnostic(std::sync::Arc<dyn Diagnostic>);
+
+impl CompileDiagnostic {
+    pub fn report<T>(db: &dyn Db, diagnostic: T)
```

Comment:
is `report` the best verb here? It feels to me like we're adding a diagnostic to be reported later, rather than reporting it to the user immediately